### ISSUE TITLE
Discussion Forum Prediction

### ISSUE THREAD

**Issue by aryantiwari10** [prathimacode-hub/ML-ProjectKart#555]
Define You:

- [x] LGM-SOC'21 Participant 
- [ ] DevIncept Participant
- [x] Contributor

Project Title : Discussion Forum Prediction

Task : To predict about the discussion forums in detail.

In this project first we first performed a exploratory data analysis on the given dataset which includes of data cleaning , data manipulation, data preprocessing , data visualization and after that we used NLP for the stopwords removal and cleaning, after using NLP we did the model building using different machine learning classification and regression algorithms and then predicted the accuracy of every model . In the model prediction part we used different machine learning algorithms . In each algorithm we had included the accuracy score , training score , confusion matrix, classification report . While in the EDA part I have included different plots for the different visualizations of our dataset .I performed the ANN and feature scaling then, keeping epochs 200 and used different activation function and the optimizers during ANN prediction, During training and testing part we included classification models such as Decision Tree Classifer , Random Forest Classifier , Ada Boost Classifier model building . While the other model accuracies can be increased more using the hypertuning . Some plots which we used for visualizing the dataset are Histogram , Barplot , Boxplot, Heatmap , Scatter plot , Pairplot , Jointplot etc.

 The Libraries used are:

PANDAS

NUMPY

MATPLOTLIB

SEABORN

SCIPY

SKLEARN

NLTK

TENSORFLOW

KERAS

**Comment by prathimacode-hub** [comment]
Issue assigned. @aryantiwari10
